File: backend/src/common/models/detection.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use crate::common::models::device::Device;

#[derive(Deserialize, Serialize, sqlx::FromRow, Debug)]
pub struct Detection {
    pub id: i64,
    pub device: Device,
    pub source: String,
    pub timestamp: DateTime<Utc>,
}

File: frontend/types/detection.ts
import {Device} from "~/types/device";

export class Detection {
  id: number;
  device: Device;
  source: string;
  timestamp: string;

  constructor(id: number, device: Device, source: string, timestamp: string) {
    this.id = id;
    this.device = device;
    this.source = source;
    this.timestamp = timestamp;
  }
}
